contracts/contracts/MACI.sol

```diff
@@ -219,15 +214,6 @@ contract MACI is IMACI, DomainObjs, Params, SnarkCommon, Ownable {
             _initialVoiceCreditProxyData
         );
 
-        // The limit on voice credits is 2 ^ 32 which is hardcoded into the
-        // MessageValidator circuit, specifically at check that there are
-        // sufficient voice credits (using GreaterEqThan(32)).
-        // TODO: perhaps increase this to 2 ^ 50 = 1125899906842624?
-        require(
```

Comment:
Our suggestion here was to be consistent across the codebase. However, considering the gas constraints, are you sure you want to unrestrict the number of voice credits? There also a valid argument that such enforcement can be done within the front-end and removing the check within the contract actually saves gas. Is that the reasoning for applying this fix?

Reply:
I think the limit for the voice credit is: `voiceCredit<sqrt(field_size)=147946756881789319005730692170996259609`
And this is larger than most (probably all) of the total hard cap of ERC20 token. (e.g. 1B tokens cap with 18 decimal)

BTW, what is the argument here?

Reply:
The argument is that you can either enforce this directly within the contract or remove the require statement and enforce it in the front-end and add documentation for developers who will rely on these contracts. The former is a more secure but the later saves a lot of gas. In either choice, the suggestion was to be more consistent with this and to properly document it.